src/proper_sa.erl

```diff
@@ -76,33 +77,36 @@
 %% Records
 %% -----------------------------------------------------------------------------
 
--record(sa_target,
-        {first             = null :: proper_types:type(),
-         next              = null :: fun((_, _) -> proper_types:type()),
-         current_generated = null :: proper_gen:instance(),
-         last_generated    = null :: proper_gen:instance()}).
+-record(
+   sa_target,
+   {first             = null :: proper_types:type(),
+    next              = null :: fun((_, _) -> proper_types:type()
+                                                | proper_gen:instance()),
```

Comment:
I think it's better to declare a type for `proper_types:type() | proper_gen:instance()` and use it here.

This will also allow to undo the change in the formatting of the two record, which adds, unnecessarily, many lines of differences in this PR.